tables/DataMan/DataManInfo.cc

```diff
@@ -207,6 +208,206 @@ Record DataManInfo::adjustStMan (const Record& dminfo, const String& dmType,
   return newdm;
 }
 
+void DataManInfo::mergeInfo (Record& dminfo1, const Record& dminfo2)
+{
+  // See for each new data manager what to do.
+  for (uInt i2=0; i2<dminfo2.nfields(); ++i2) {
+    Record dm2 = dminfo2.subRecord(i2);
+    String type2;
+    String name2;
+    Vector<String> cols2;
+    if (dm2.isDefined("TYPE")) {
+      type2 = dm2.asString ("TYPE");
+    }
+    if (dm2.isDefined("NAME")) {
+      name2 = dm2.asString ("NAME");
+    }
+    // Add the data manager to the first, but overwrite if already there.
+    uInt inx = dminfo1.nfields();
+    for (uInt i1=0; i1<dminfo1.nfields(); ++i1) {
+      const Record& dm1 = dminfo1.subRecord(i1);
+      String type1;
+      String name1;
+      // An empty or undefined type/name means use the other.
+      if (dm1.isDefined("TYPE")) {
+        type1 = dm1.asString ("TYPE");
+      }
+      if (dm1.isDefined("NAME")) {
+        name1 = dm1.asString ("NAME");
+      }
+      if (type1.empty()) {
+        type1 = type2;
+      }
+      if (name1.empty()) {
+        name1 = name2;
+      }
+      if (type2.empty()) {
+        type2 = type1;
+      }
+      if (name2.empty()) {
+        name2 = name1;
+      }
+      if (type1==type2 && name1==name2) {
+        inx = i1;
+        // Use the old specs if undefined in new one.
+        if (!dm2.isDefined("SPEC")  &&  dm1.isDefined("SPEC")) {
+          dm2.defineRecord ("SPEC", dm1.subRecord("SPEC"));
+        }
+        mergeColumns (dminfo1, inx, dm2);
+        break;
+      }
+    }
+    dminfo1.defineRecord (inx, dm2);
+  }
+}
+
+void DataManInfo::mergeColumns (Record& dminfo, uInt inxdm, Record& dm)
+{
+  // Get the columns from the new dm.
+  Vector<String> cols;
+  if (dm.isDefined("COLUMNS")) {
+    cols.reference(dm.asArrayString ("COLUMNS"));
+  }
+  if (! cols.empty()) {
+    // Iterate over all dm-s.
+    for (uInt i=0; i<dminfo.nfields(); ++i) {
+      Record dm2 = dminfo.subRecord(i);
+      if (dm2.isDefined("COLUMNS")) {
+        Vector<String> cols2(dm2.asArrayString("COLUMNS"));
+        if (! cols2.empty()) {
+          std::vector<String> colsnew;
+          // Keep column if not equal to one in new dm.
+          for (auto col2 : cols2) {
+            Bool fnd = False;
+            for (auto col : cols) {
+              if (col2 == col) {
+                fnd = True;
+                break;
+              }
+              if (!fnd) {
```

Comment:
I don't understand the logic, but it looks like you should move this `if` out of the inner for loop.